modules/systemd-journal/journal-reader.c

```diff
@@ -233,6 +233,7 @@ _set_program(JournalReaderOptions *options, LogMessage *msg)
     }
   else
     {
+      g_free(value);
```

Comment:
I refactored a little bit this part of the code, the refactored one is simpler than this. Opened a pr to your branch, please merge.
And thanks for the leak-fixes!

Reply:
Thanks, merged.